feat: ramp analytics segment migration

NicolasMassart opened the pull request:
## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

Use the new Segment metametrics hook in the ramp specific analytics hook

- update the hook
- rework the anonymous tracking logic as it's handled on the useMetrics hook

## **Related issues**

Fixes #8455

## **Manual testing steps**

1. On wallet touch "Buy ETH" button
2. Wait for the onramp screen to complete (select region if needed)
3. touch "cancel" on top right
4. check the logs (fi in dev mode) you must see: `TRACK event saved {"event": "On-ramp Canceled", "properties": {"chain_id_destination": "1", "location": "Amount to Buy Screen"}, "type": "track"}`
5. check the segment source debugger and you must see the following
    <img width="1158" alt="image" src="https://github.com/MetaMask/metamask-mobile/assets/4677568/95a86b69-2691-4e8a-98ef-60d51281e8ba">

## **Screenshots/Recordings**

<img width="300" alt="screenshot" src="https://github.com/MetaMask/metamask-mobile/assets/4677568/cbdf41b7-4e9c-42f8-8d14-7de9a23fca3a">

### **Before**

<!-- [screenshots/recordings] -->
NA

### **After**

<!-- [screenshots/recordings] -->
NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

NicolasMassart committed:
use metrics in ramp analytics

NicolasMassart committed:
restore InteractionManager.runAfterInteractions

NicolasMassart committed:
remove JsonMap dependency

NicolasMassart committed:
use if instead of ternary

wachunei reviewed (dismissed):
Clean! LGTM!

NicolasMassart committed:
WIP on testing hook anon event handling

NicolasMassart committed:
Merge branch 'main' into feat/batch_1129_segment_2

NicolasMassart committed:
remove redundant var

NicolasMassart committed:
Merge branch 'main' into feat/batch_1129_segment_2

Cal-L reviewed (commented):
Looks like both unit and e2e test is failing. Is the E2E failure related to your changes?

NicolasMassart committed:
Remove test

as it requires Analytics to be remove before to allow mocking
Will introduce it again at the end

NicolasMassart commented:
> Looks like both unit and e2e test is failing. Is the E2E failure related to your changes?

I don't think it's related at all
